src/autosuspend/checks/activity.py

```diff
@@ -693,7 +700,14 @@ def __init__(self, name: str, types: Iterable[str], states: Iterable[str]) -> No
         self._states = states
 
     def check(self) -> Optional[str]:
-        for session_id, properties in list_logind_sessions():
+        try:
```

Comment:
Could you please wrap this repeated code fragment in a utility function inside this file

Reply:
😕 having to pass the logger in is a bit clunky. Would you prefer it to just not log?